clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2232,7 +2250,8 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
     const ParmVarDecl *SamplerArg = InitMethod->getParamDecl(0);
     assert(SamplerArg && "sampler __init method must have sampler parameter");
 
-    addParam(FD, SamplerArg->getType(), SYCLIntegrationHeader::kind_sampler);
+    addParam(FD, SamplerArg->getType(), SYCLIntegrationHeader::kind_sampler,
```

Comment:
Can you add a comment explaining why this specialization was needed?

Reply:
Done!